include/fluent-bit/flb_output.h

```diff
@@ -676,21 +676,13 @@ struct flb_output_flush *flb_output_flush_create(struct flb_task *task,
                                     evc->tag, flb_sds_len(evc->tag),
                                     evc->data, evc->size,
                                     &p_buf, &p_size);
-            if (ret == -1 || p_size == 0) {
+            if (ret == -1) {
                 flb_coro_destroy(coro);
                 flb_free(out_flush);
                 return NULL;
             }
 
-            records = flb_mp_count(p_buf, p_size);
-            if (records == 0) {
-                flb_coro_destroy(coro);
-                flb_free(out_flush);
-                flb_free(p_buf);
-                return NULL;
-            }
-
-            tmp = flb_event_chunk_create(evc->type, records, evc->tag, flb_sds_len(evc->tag), p_buf, p_size);
+            tmp = flb_event_chunk_create(evc->type, flb_mp_count(p_buf, p_size), evc->tag, flb_sds_len(evc->tag), p_buf, p_size);
```

Comment:
flb_mp_count() can fail if there is a corrupted record, it's a bit safer to validate its return value out of the function. For the majority of use cases except for sds_len() we do not use function returns as arguments

Reply:
@edsiper sure, I can fix.